Whiskey/Functions/Resolve-WhiskeyTaskPathParameter.ps1

```diff
@@ -0,0 +1,143 @@
+        [object]$TaskContext,
+
+        [Parameter(ValueFromPipeline=$true)]
+        [string]$Path,
+
+        [Parameter(Mandatory)]
+        [string]$PropertyName,
+
+        [Parameter(Mandatory)]
+        [Management.Automation.ParameterMetadata]$CmdParameter,
+
+        [Parameter(Mandatory)]
+        [Whiskey.Tasks.ValidatePathAttribute]$ValidateAsPathAttribute
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx = -1
+    }
+
+    process
+    {
+        Set-StrictMode -Version 'Latest'
+
+        $pathIdx++
+
+        $result = $Path
+
+        if( -not $Path )
+        {
+            if( $ValidateAsPathAttribute.Mandatory )
+            {
+                Stop-WhiskeyTask -TaskContext $Context `
+                                 -PropertyName $CmdParameter.Name `
+                                 -Message ('{0} is mandatory.' -f $TaskProperty[$propertyName])
+                return
+            }
+            return     
+        }
+
+        if( -not [IO.Path]::IsPathRooted($Path) )
+        {
+            $Path = Join-Path -Path $TaskContext.BuildRoot -ChildPath $Path
+        }
+
+        $optionalParams = @{}
+        if( -not $ValidateAsPathAttribute.MustExist)
+        {
+            $optionalParams['ErrorAction'] = 'Ignore'
+        }
+
+        $message = 'Resolve {0} ->' -f $result
+        $prefix = ' ' * ($message.Length - 3)
+        Write-Debug -Message $message
+        $result = 
+            Resolve-Path -Path $Path @optionalParams | 
+            Select-Object -ExpandProperty 'ProviderPath' |
+            ForEach-Object { 
+                Write-Debug -Message ('{0} -> {1}' -f $prefix,$_)
+                $_
+            }
+
+        if( -not $result ) 
+        {
+            if( $ValidateAsPathAttribute.MustExist )
+            {
+                Stop-WhiskeyTask -TaskContext $TaskContext `
+                                 -Message ('{0}[{1}] "{2}" does not exist and must exist.' -f $PropertyName,$pathIdx,$Path,$TaskContext.ConfigurationPath)
+                return
+            }
+            $result = [IO.Path]::GetFullPath($Path)
+        }
+            
+        if( -not $ValidateAsPathAttribute.AllowOutsideBuildRoot )
+        {
+            $normalizedBuildRoot = $TaskContext.BuildRoot.FullName.TrimEnd([IO.Path]::DirectorySeparatorChar, [IO.Path]::AltDirectorySeparatorChar)
+            $normalizedBuildRoot = '{0}{1}' -f $normalizedBuildRoot,[IO.Path]::DirectorySeparatorChar
+            
+            $invalidPaths =
+                $result |
+                Where-Object { -not ( $_.StartsWith($normalizedBuildRoot) ) }
+
+            if( $invalidPaths )
+            {
+                Stop-WhiskeyTask -TaskContext $TaskContext `
+                                 -Message ('{0}[{1}] "{2}" is outside of the build root and not allowed to be.' -f $PropertyName,$pathIdx,$Path,$TaskContext.ConfigurationPath)

```

Comment:
Shorten to just `is outside the build root "{3}"` and include what that is in the message.